### PR TITLE
Fix using of @php scripts in autoscripts

### PR DESCRIPTION
**Pull request by alexander-schranz**
This will **not** fix other @ commands like reference calls:

```json
    "scripts": {
        "auto-scripts": [
            "@test-scripts"
        ],
        "test-scripts": [
        ],
     }
```

and also **not** fixing running `@composer` commands:

```json
    "scripts": {
        "auto-scripts": [
            "@composer something"
        ]
     }
```

fixes parts of #431

### CODE REVIEW

File: src/ScriptExecutor.php [hunk 1]
```diff
@@ -43,12 +43,17 @@ public function __construct(Composer $composer, IOInterface $io, Options $option
...
+            $type = 'php-script';
+        }
+
         $parsedCmd = $this->options->expandTargetDir($cmd);
         if (null === $expandedCmd = $this->expandCmd($type, $parsedCmd)) {
             return;
         }
 
-        $cmdOutput = new StreamOutput(fopen('php://temp', 'rw'), OutputInterface::VERBOSITY_VERBOSE, $this->io->isDecorated());
+        $cmdOutput = new StreamOutput(fopen('php://temp', 'rwb'), OutputInterface::VERBOSITY_VERBOSE, $this->io->isDecorated());
```
Comment (alexander-schranz) [inline src/ScriptExecutor.php]: this was changed by the code style fixer

**Comment by alexander-schranz** [comment]
@nicolas-grekas not sure how to fix this issue as there are other cases which currnetly don't work as `@php`. The could be `@composer` or `@customscript` but don't know how to call composer directly with it.

**Comment by alexander-schranz** [comment]
@fabpot I don't think so when a look at the [composer documentation](https://getcomposer.org/doc/articles/scripts.md#command-events)

**Comment by fabpot** [comment]
ok, so I don't understand what we are trying to fix here.

**Comment by alexander-schranz** [comment]
@fabpot we did run into this problems on a project which doesn't use flex but a developer had flex installed globally and got this [strange error message](https://github.com/symfony/flex/blob/ef8b60eed1d51aae05feb79bb60edc52f55c8142/src/ScriptExecutor.php#L89).

At least we should change the error message that `flex` does not support running this type of scripts. In our errors case supporting for `@php` would be enough but just wanted to mention that there are also other cases which are documented in the composer docs which don't work when used inside auto-scripts in symfony flex documentation.

**Comment by fabpot** [comment]
`@auto-scripts` has a special syntax that you should use. Don't use it with the "standard" syntax from Composer (for which you can use another alias).

**Pull request closed without merging**